Initial tests

katie291100 opened the pull request:
resolves #1 

Description: Create the first test class (utilitiesTest) and learn how to run it etc. Also create test run configuration 

To do (If relevant): 

Reviewing responsibility: @Cdsspnks96

Commit:
Working on setting up tests (not done).

Commit:
Tests now work.
Added configuration file to run all tests in the project.

Commit:
First test of utilities to help with understanding

Commit:
resolves #4
create QueueFIFOTest

Commit:
Added tests for half of the QueueFIFO class.
Also added a get(), set() and getI() class in QueueFIFO for testing purposes.

Cdsspnks96 reviewed (approved):
All changes functional and hence approved.

Cdsspnks96 committed:
VectorToAngle Test Implemented

Cdsspnks96 committed:
AngleToVector Test Implemented

Commit:
Utilities tests.
Implement tests for:
- round
- contains
Amends test for:
- floor

Commit:
Completed tests for all implemented methods in QueueFIFO

Commit:
Clean Up

Commit:
checkProxmity Test Implemented
Additionally fixed checkProximity bug when distance between vectors is identical to the allowed radius.

Commit:
tilesToDistance and distanceToTiles Tests Implemented

Commit:
Remove redundant imports

Cdsspnks96 committed:
Implemented pop() test in QueueFIFO
Fixed pervasive issue regarding topIndex being larger than the highest possible index in Queue (size() rather than size()-1) and amended existing tests accordingly.

Cdsspnks96 committed:
angleBetween Test Implemented

Cdsspnks96 committed:
scale Test Implemented and messages added to assert statements

Cdsspnks96 committed:
Improved angleBetween Test and fixed inaccuracy issue with angleBetween method

Cdsspnks96 committed:
Removed redundant property of randomChoice

Cdsspnks96 committed:
Implemented test for testScale, renamed scale and testScale to scaleTwoVectors and scaleFourPoints to more easily differentiate which test is for which method

Cdsspnks96 committed:
Implemented remaining tests in QueueFIFOTest

Cdsspnks96 reviewed (commented):
New changes since last review are functional, though there are some tests that don't contain error messages.
